### PR TITLE
Deletion Performance Optimization

### PR DESCRIPTION
**Pull request by czaloom**
# Performance Optimizations for Resource Constrained Postgresql

## Changes
- ~timeout controls exposed in client~ #660 
- ~evaluation response creation no longer spams postgres~ #661
- deletion in the backend is now completely manual

### CODE REVIEW

File: api/tests/functional-tests/backend/core/test_evaluation.py [hunk 1]
```diff
@@ -885,67 +884,3 @@ def test_count_active_evaluations(
         )
         == 0
     )
-
-
-def test__fetch_evaluations_and_mark_for_deletion(
```
Comment (ekorman) [inline api/tests/functional-tests/backend/core/test_evaluation.py:890]: do we have coverage elsewhere for what this was testing?

File: api/valor_api/backend/core/annotation.py [hunk 2]
```diff
@@ -358,18 +358,37 @@ def delete_dataset_annotations(
...
+        select(models.Annotation)
         .join(models.Datum, models.Datum.id == models.Annotation.datum_id)
         .where(models.Datum.dataset_id == dataset.id)
         .subquery()
     )
-    delete_stmt = delete(models.Annotation).where(
-        models.Annotation.id == subquery.c.id
-    )
 
+    # delete embeddings (if they exist)
```
Comment (ekorman) [inline api/valor_api/backend/core/annotation.py:368]: this block of comments should be deleted right?

File: api/valor_api/backend/core/annotation.py [hunk 3]
```diff
@@ -401,17 +420,36 @@ def delete_model_annotations(
...
+    annotations_to_delete = (
+        select(models.Annotation)
         .where(models.Annotation.model_id == model.id)
         .subquery()
     )
-    delete_stmt = delete(models.Annotation).where(
-        models.Annotation.id == subquery.c.id
-    )
 
+    # # delete embeddings (if they exist)
```
Comment (ekorman) [inline api/valor_api/backend/core/annotation.py:429]: this block too

File: api/valor_api/backend/core/datum.py [hunk 4]
```diff
@@ -251,3 +252,38 @@ def get_paginated_datums(
     )
 
     return (content, headers)
+
+
+def delete_datums(
```
Comment (ekorman) [inline api/valor_api/backend/core/datum.py:257]: can you add a funcitonal-test for this?

File: api/tests/functional-tests/backend/core/test_evaluation.py (hunk 1 above)
Comment (czaloom) [reply api/tests/functional-tests/backend/core/test_evaluation.py:890]: The function this was testing was rolled into the `delete_evaluations` function in `backend/core/evaluation.py`.

File: api/valor_api/backend/core/datum.py (hunk 4 above)
Comment (czaloom) [reply api/valor_api/backend/core/datum.py:257]: This should be covered by this [test](https://github.com/Striveworks/valor/blob/e2152b120173fb0550074db63796de6cbd39f68d/api/tests/functional-tests/crud/test_create_delete.py#L461-L467).

File: api/tests/functional-tests/backend/core/test_evaluation.py (hunk 1 above)
Comment (czaloom) [reply api/tests/functional-tests/backend/core/test_evaluation.py:890]: This is covered in a few places but i appended the test to the crud tests [here](https://github.com/Striveworks/valor/pull/659/commits/9930a144304bb5a382b92c2960b1a4d2f068dda5).

Comment (czaloom) [reply api/tests/functional-tests/backend/core/test_evaluation.py:890]: The actual mark for deletion shouldnt affect anything as deletion is only allowed for evaluations in the `Done` or `Failed` state.

Comment (czaloom) [reply api/tests/functional-tests/backend/core/test_evaluation.py:890]: added a functional test [here](https://github.com/Striveworks/valor/pull/659/commits/12fb2d447671ad04811b850ba21e215b0c2b2737)

**Review by ekorman — APPROVED** [review]

**Pull request merged**